hoomd/hpmc/pytest/test_convex_polyhedron.py

```diff
@@ -1,32 +1,146 @@
+              'sweep_radius': 2.0}
+    test_convex_polyhedron3 = _hpmc.PolyhedronVertices(args_3)
     test_dict3 = test_convex_polyhedron3.asDict()
     assert test_dict3 == args_3
-    
-    args_4 = {'vertices':[(0, 1, 0), (1, 1, 1), (1, 0, 1), (0, 1, 1), (1, 1, 0), (0, 0, 1), (0, 0, 1), (0, 0, 1)], 'ignore_statistics':0}
-    test_convex_polyhedron4 = hpmc.poly3d_verts(args_4)
+
+    args_4 = {'vertices': [(0, 1, 0), (1, 1, 1), (1, 0, 1), (0, 1, 1),
+                           (1, 1, 0), (0, 0, 1), (0, 0, 1), (0, 0, 1)],
+              'ignore_statistics': 0,
+              'sweep_radius': 4.0}
+    test_convex_polyhedron4 = _hpmc.PolyhedronVertices(args_4)
     test_dict4 = test_convex_polyhedron4.asDict()
     assert test_dict4 == args_4
-    
-    args_5 = {'vertices':[(0, 10, 3), (3, 2, 1), (1, 2, 1), (0, 1, 1), (1, 1, 0), (5, 0, 1), (0, 10, 1), (9, 5, 1), (0, 0, 1)], 'ignore_statistics':1}
-    test_convex_polyhedron5 = hpmc.poly3d_verts(args_5)
+
+    args_5 = {'vertices': [(0, 10, 3), (3, 2, 1), (1, 2, 1), (0, 1, 1),
+                           (1, 1, 0), (5, 0, 1), (0, 10, 1), (9, 5, 1),
+                           (0, 0, 1)],
+              'ignore_statistics': 1,
+              'sweep_radius': 5.5}
+    test_convex_polyhedron5 = _hpmc.PolyhedronVertices(args_5)
     test_dict5 = test_convex_polyhedron5.asDict()
     assert test_dict5 == args_5
-    
 
+
+# these tests are for the python side
+def test_convex_polyhedron_python():
+
+    verts = [(-1, 1, 0), (1, 0, -1), (1, 1, 1), (-1, -1, 1)]
+    poly = hpmc.integrate.ConvexPolyhedron(23456)
+    poly.shape['A'] = dict(vertices=verts)
+    assert not poly.shape['A']['ignore_statistics']
+    np.testing.assert_allclose(poly.shape['A']['vertices'], verts)
+
+
+def test_convex_polyhedron_after_attaching(device, dummy_simulation_factory):
+
+    verts = [(-1, 1, 0), (1, 0, -1), (1, 1, 1), (-1, -1, 1)]
+    verts2 = [(-1, 1, 1), (1, -1, 0), (1, 1, 1)]
+    poly = hpmc.integrate.ConvexPolyhedron(23456)
+    poly.shape['A'] = dict(vertices=verts)
+    poly.shape['B'] = dict(vertices=verts2, ignore_statistics=True)
+
+    sim = dummy_simulation_factory(particle_types=['A', 'B'])
+    sim.operations.add(poly)
+    sim.operations.schedule()
+
+    assert not poly.shape['A']['ignore_statistics']
+    assert poly.shape['B']['ignore_statistics']
+    np.testing.assert_allclose(poly.shape['A']['vertices'], verts)
+    np.testing.assert_allclose(poly.shape['B']['vertices'], verts2)
+
+
+def test_overlaps(device, dummy_simulation_check_overlaps):
+
+    mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456, d=0, a=0)
+    mc.shape['A'] = dict(vertices=[(0, (0.75**0.5) / 2, -0.5),
+                                   (-0.5, -(0.75**0.5) / 2, -0.5),
+                                   (0.5, -(0.75**0.5) / 2, -0.5),
+                                   (0, 0, 0.5)])
+    sim = dummy_simulation_check_overlaps()
+    sim.operations.add(mc)
+    # gsd_dumper = hoomd.dump.GSD(filename='/Users/danevans/hoomd/
```

Comment:
The above comments to `test_convex_polygon.py` apply to all these tests, except that 3D shapes should of course initialize 3-d systems for testing.